TIGLCreator/src/TIGLCreatorSelectWingAndFlapStatusDialog.cpp

```diff
@@ -165,94 +197,99 @@ void TIGLCreatorSelectWingAndFlapStatusDialog::buildFlapRow(const tigl::CCPACSTr
 
     QString rot;
 
-    if ( controlSurfaceDevice.GetPath().GetSteps().GetSteps().size() > 0 ) {
-        const auto& step = controlSurfaceDevice.GetPath().GetSteps().GetSteps().front();
-        if(step) {
+    if (controlSurfaceDevice->GetPath().GetSteps().GetSteps().size() > 0) {
+        const auto& step = controlSurfaceDevice->GetPath().GetSteps().GetSteps().front();
+        if (step) {
             rot.append(QString::number(step->GetHingeLineRotation().value_or(0.)));
         }
     }
 
-    QLabel* labelRotation   = new QLabel(rot, this);
+    QLabel* labelRotation = new QLabel(rot, this);
     labelRotation->setStyleSheet(transparentBG);
 
     gridLayout->setCellWidget(rowIdx, 3, labelRotation);
 
-    double savedValue = controlSurfaceDevice.GetControlParameter();
+    double savedValue = controlSurfaceDevice->GetControlParameter();
+
+    double minControlParam = controlSurfaceDevice->GetMinControlParameter();
+    double maxControlParam = controlSurfaceDevice->GetMaxControlParameter();
 
-    double minControlParam = controlSurfaceDevice.GetMinControlParameter();
-    double maxControlParam = controlSurfaceDevice.GetMaxControlParameter();
-    
-    int newSliderValue = static_cast<int>((slider->maximum() - slider->minimum()) / (maxControlParam-minControlParam) * (savedValue - minControlParam))
-                        + slider->minimum();
+    int newSliderValue = static_cast<int>((slider->maximum() - slider->minimum()) /
+                                          (maxControlParam - minControlParam) * (savedValue - minControlParam)) +
+                         slider->minimum();
     slider->setValue(newSliderValue);
 
     spinBox->setMinimum(minControlParam);
     spinBox->setValue(savedValue);
     spinBox->setMaximum(maxControlParam);
-    
+
     DeviceWidgets elements;
-    elements.slider = slider;
-    elements.controlParamBox = spinBox;
-    elements.rotAngleLabel = labelRotation;
+    elements.slider            = slider;
+    elements.controlParamBox   = spinBox;
+    elements.rotAngleLabel     = labelRotation;
     _guiMap[uid.toStdString()] = elements;
 
-    slider->setWindowTitle( uid );
-    spinBox->setWindowTitle( uid );
+    slider->setWindowTitle(uid);
+    spinBox->setWindowTitle(uid);
 
     connect(slider, SIGNAL(valueChanged(int)), this, SLOT(slider_value_changed(int)));
     connect(spinBox, SIGNAL(valueChanged(double)), this, SLOT(spinBox_value_changed(double)));
 }
 
 void TIGLCreatorSelectWingAndFlapStatusDialog::drawGUI()
 {
-    cleanup();    
+    cleanup();
+
     std::string wingUID = m_currentWing;
 
     if (wingUID.empty())
         return;
 
     tigl::CCPACSConfiguration& config = _document->GetConfiguration();
-    tigl::CCPACSWing &wing = config.GetWing(wingUID);
+    tigl::CCPACSWing& wing            = config.GetWing(wingUID);
 
-    std::vector<tigl::CCPACSTrailingEdgeDevice*> devices;
+    std::vector<tigl::CTiglAbstractGeometricComponent*> devices;
```

Comment:
A small code refactoring suggestion: Instead of using pointers to a too broad base class here and then using dynamic_cast to check the actual type, we could use std::variant, because we don't expect there to be a very large number of different control devices in the future (spoilers are the only type that is currently on plan).

https://godbolt.org/z/jbK9Mcs9d

You can pass functions like buildFlapRow directly to std::visit. You could use templated helper methods or lambdas with auto argument to get the uids or control parameters and pass those to `std::visit`.